meta-ostro/classes/ostro-image.bbclass

```diff
@@ -381,6 +382,39 @@ inherit xattr-images
 # Create all users and groups normally created only at runtime already at build time.
 inherit systemd-sysusers
 
+# Provide an image feature that disables all consoles and makes
+# journald mute.
+python() {
+    if bb.utils.contains('IMAGE_FEATURES', 'muted', True, False, d):
+        import re
+
+	# Mangle cmdline to drop all consoles and add quiet option
+        cmdline = d.getVar('APPEND', True)
+        new=re.sub('console=\w+(,\w*)*\s','', cmdline) + ' quiet'
+        d.setVar('APPEND', new)
+}
+
+# In addition to cmdline mangling, make sure journal only gets
+# emergency errors and any lower priority messages do not get
+# logged. This is temporary approach and the cleanest way to disable
+# logging (compared with removing the journald binary and related
+# services here). Once the systemd recipe gives us better granulatiry
+# for packaging, the preferred way to avoid logging is
+# to not install systemd-journald at all.
+muted () {
+    sed -i -e 's/^#\(MaxLevelStore=\).*/\1emerg/'\
+           -e 's/^\(ForwardToSyslog=yes\)/#\1/' \
+           ${IMAGE_ROOTFS}${sysconfdir}/systemd/journald.conf
+
+    # Remove systemd-getty-generator to avoid (serial-)getty services
+    # being created for kernel detected consoles.
+    rm ${IMAGE_ROOTFS}${systemd_unitdir}/system-generators/systemd-getty-generator
```

Comment:
why not do it just in the kcmdline? https://www.freedesktop.org/software/systemd/man/systemd-journald.service.html says you can use `systemd.journald.forward_to_syslog=0` and `systemd.log_level=emerg` (but this one needs to be checked, may not be used by journald).